File: 2021/src/test/java/nel/marco/Day9Test.java
package nel.marco;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day9Test {

    Day9 day9 = new Day9();

    public List<String> readInput(boolean readExample) throws IOException {

        String path = "src\\main\\resources\\day9";
        if (readExample) {
            path = "src\\main\\resources\\day9Example";
        }

        List<String> stringList = Files.readAllLines(Path.of(path));

        return new ArrayList<>(stringList);

    }


    @Test
    void part1() throws IOException {

        assertEquals(15, day9.part1(readInput(true)));
        assertEquals(458, day9.part1(readInput(false)));
    }

    @Test
    void part2() throws IOException {

        assertEquals(1134, day9.part2(readInput(true)));
        assertEquals(1391940, day9.part2(readInput(false)));
    }
}

File: 2024/src/test/nel/marco/Day9Test.kt
package nel.marco

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotEquals
import org.junit.jupiter.api.Test

class Day9Test {

    val day = Day9(useExample = false, useMac = true)
    val example = Day9(useExample = true, useMac = true)

    @Test
    fun answerOne() {
        assertEquals("1928", example.answerOne())
        assertEquals("6288707484810", day.answerOne()) // not 6273684533700
    }

    @Test
    fun answerTwo() {
        assertEquals("2858", example.answerTwo())
        assertEquals("6311837662089", day.answerTwo()) // cheated this one :(
    }

    fun avg() {
        val day = day
        MarcoUtil.avgTime("part1", 5) {
            assertEquals("6288707484810", day.answerOne())
        }
        MarcoUtil.avgTime("part2", 5) {
            assertEquals("X", day.answerTwo())
        }
    }
}

